android/javatests/org/chromium/chrome/browser/BytecodeTest.java

```diff
@@ -291,6 +291,9 @@ public void testMethodsExist() throws Exception {
         Assert.assertTrue(methodExists(
                 "org/chromium/chrome/browser/search_engines/settings/SearchEngineAdapter",
                 "sortAndFilterUnnecessaryTemplateUrl", false, null));
+        Assert.assertTrue(methodExists(
+                "org/chromium/chrome/browser/search_engines/settings/SearchEngineAdapter",
+                "didSearchEnginesChange", false, null));
```

Comment:
We should also make sure that `containsTemplateUrl` and `computeStartIndexForRecentSearchEngines` still exist too.

Reply:
done for `computeStartIndexForRecentSearchEngines`:
https://github.com/brave/brave-core/pull/14402/commits/22d3cd2546b3d11d11a1785792e4d6482e83dee8#diff-b2d64845479ca6df40c36fb252689f1ebf820afd2e1b2f80bd684e489ab64f38R299

and `containsTemplateUrl` went away because I call `didSearchEnginesChange` via reflection at
https://github.com/brave/brave-core/pull/14402/commits/22d3cd2546b3d11d11a1785792e4d6482e83dee8#diff-0ad610dfe8c1500950c13c42ab4758c5f1d8863ba25cf52c8a194c3d9d5a36daR91

Reply:
One more thing here, we need to move these checks to `testMethodsForInvocationExist` with checking exact types as they may change it in upstream and we will have troubles on trying to invoke those methods. Other than that looks good!